### PR TITLE
armpl-gcc: Add version 22.1

### PR DESCRIPTION
**Pull request by annop-w**
- Add version 22.1
- Modify default url to point to version 22.1

### CODE REVIEW

**Comment by annop-w** [comment]
@alalazo Could you please review this too ? Thank you.

**Comment by haampie** [comment]
Why is the GCC version relevant here?

**Comment by annop-w** [comment]
There are multiple versions of armpl compiled with different version of gcc. Therefore, we try to have users installed the version that is compatible with their gcc version.

File: var/spack/repos/builtin/packages/armpl-gcc/package.py [hunk 1]
```diff
@@ -123,6 +154,12 @@ class ArmplGcc(Package):
     conflicts("target=ppc64:", msg="Only available on Aarch64")
     conflicts("target=ppc64le:", msg="Only available on Aarch64")
 
+    conflicts("%gcc@:11.0", when="@22.1_gcc-11.2")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/armpl-gcc/package.py]: Do you mean to say conflicts with GCC version `< 11.0`? Or `<= 11.0.x`?

Comment (annop-w) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: < 11.0.

Comment (haampie) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: In that case it's `@:10`, this upperbound _includes_ any `10.x` version.

Comment (annop-w) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: Done. I've also fixed a url issue.

**Review by haampie — APPROVED** [review]

**Pull request merged**